src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +44,40 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+
+        StringBuilder resultBuilder = new StringBuilder();
+        resultBuilder.append("Отчёт за период ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String myEmployeeName : names) {
+            int earnedMoney = 0;
+            for (String myWorkingData : data) {
+
+                String[] dataArray = myWorkingData.split(" ");
+                int beginDate = convertStringDate(dateFrom.split("\\."));
+                int endDate = convertStringDate(dateTo.split("\\."));
+                int currentDate = convertStringDate(dataArray[0].split("\\."));
+
+                if (beginDate > endDate) {
+                    throw new IllegalDateParametersException("Wrong parameters");
+                }
+
+                if (beginDate <= currentDate
+                        && endDate >= currentDate
+                        && myEmployeeName.equalsIgnoreCase(dataArray[1])) {
+                    earnedMoney += Integer.parseInt(dataArray[2]) * Integer.parseInt(dataArray[3]);
+                }
+            }
+            resultBuilder.append("\n").append(myEmployeeName).append(" - ").append(earnedMoney);
+        }
+        return resultBuilder.toString();
+    }
+
+    //Doing reverse for array with date strings to int.
```

Comment:
read smth about LocalDate and DateTimeFormatter
using int isn't correct in this case

Reply:
Ok.